src/components/jira/JiraStatus.tsx

```diff
@@ -0,0 +1,160 @@
+import { Service } from "io-functions-commons/dist/generated/definitions/Service";
+import React, { Component, Fragment } from "react";
+import { WithNamespaces, withNamespaces } from "react-i18next";
+import { Alert } from "reactstrap";
+import {
+  getServiceReviewStatus,
+  ServiceReviewStatusResponse,
+  ServiceStatus
+} from "../../utils/service";
+
+type OwnProps = {
+  t: (key: string) => string;
+  service: Service;
+  onLoaded: (event: ServiceReviewStatusResponse) => void;
+};
+
+type Props = WithNamespaces & OwnProps;
+
+class JiraStatus extends Component<Props> {
+  public state: {
+    status: ServiceStatus;
+  } = {
+    status: ServiceStatus.LOADING
+  };
+
+  public componentDidMount() {
+    getServiceReviewStatus(this.props.service)
+      .then(res => {
+        this.props.onLoaded(res);
+        this.setState({
+          status: res.status
+        });
+      })
+      .catch(() => {
+        this.setState({
+          status: ServiceStatus.NOT_FOUND
+        });
+      });
+  }
+
+  public render() {
+    const { t } = this.props;
+    const { status } = this.state;
+    return (
+      <Fragment>
+        <div className="col-12">
+          {status === ServiceStatus.LOADING ? (
+            <div className="service-status">
+              <div>
+                <span className="circle" />
+                <div>
+                  <span className="light-text">
+                    {t("service:state")}:&nbsp;
+                  </span>
+                  <span className="dark-text">
+                    {t("profile:service_loading")}
+                  </span>
+                </div>
+              </div>
+            </div>
```

Comment:
This components is almost the same for all the status. What if we render it only once instead of five?
```typescript
private getColorClass(status: ServiceStatus) {
    switch (status) {
      case "REJECTED":
        return "circle-red";
      case "REVIEW":
        return "circle-yellow";
      case "VALID":
      case "DEACTIVE":
        return "circle-green";
      default:
        return "";
    }
  }

  private getText(status: ServiceStatus) {
    switch (status) {
      case "LOADING":
        return "profile:service_loading";
      case "DRAFT":
      case "NOT_FOUND":
        return "profile:service_draft";
      case "REJECTED":
        return "profile:service_not_valid";
      case "REVIEW":
        return "profile:service_review";
      case "VALID":
      case "DEACTIVE":
        return "profile:service_valid";
      default:
        return "";
    }
  }

...

<Fragment>
        <div className="col-12">
          <div className="service-status">
            <div>
              <span className={`circle ${this.getColorClass(status)}`} />
              <div>
                <span className="light-text">{t("service:state")}:&nbsp;</span>
                <span className="dark-text">{t(this.getText(status))}</span>
              </div>
            </div>
          </div>
...
```

Reply:
fix with 1af1852

Reply:
Sorry my bad, is better use the enum instead of static strings
```typescript
private getColorClass(status: ServiceStatus) {
    switch (status) {
      case ServiceStatus.REJECTED:
        return "circle-red";
      case ServiceStatus.REVIEW:
        return "circle-yellow";
      case ServiceStatus.VALID:
      case ServiceStatus.DEACTIVE:
        return "circle-green";
      default:
        return "";
    }
  }

  private getText(status: ServiceStatus) {
    switch (status) {
      case ServiceStatus.LOADING:
        return "profile:service_loading";
      case ServiceStatus.DRAFT:
      case ServiceStatus.NOT_FOUND:
        return "profile:service_draft";
      case ServiceStatus.REJECTED:
        return "profile:service_not_valid";
      case ServiceStatus.REVIEW:
        return "profile:service_review";
      case ServiceStatus.VALID:
      case ServiceStatus.DEACTIVE:
        return "profile:service_valid";
      default:
        return "";
    }
  }
```

Reply:
Fix with 580f2ab